### DISCUSSION TITLE
status report

### DISCUSSION THREAD

**Discussion by lorbetzki** [revk/xESP32-Faikout#121] [General]
Hi,

I use a different automation than HA. I use Symcon and I have also created a module for Faikin. My problem is that I only get "updates" like current temperature, RPM etc. when I use Faikinreport.

But the faikingreport give me for some events only an array with min/avg/max, but In this case i need the "latest" value.

For example why that is bad:

1. I turn the system on, then off by mistake and then on again
-> i receive a status with on, off, on. That is OK
--> now i receive a reporting update AFTER my status and i have the choice of min = 0 and max 1. Which is the correct one? 

2. i set the temperature to 22°, then by mistake to 19° and then 21°
-> i receive a status with 22°, 19°, 21°. That is OK.
--> i receive a reporting update AFTER my status with min = 19°, avg = 20,6°, max 22°. You see my problem?

Sure I can go and get only certain values from the report, but I would be happier if there was a periodic status update. Either only with changed values or all last.

What do you think?

**Comment by revk** [comment]
--> i receive a reporting update AFTER my status with min = 19°, avg = 20,6°, max 22°. You see my problem?

That is because in the period (minute) that was the range of the setting. It will be averaged for each second I think, so depends on the timing of each level.

**Comment by lorbetzki** [comment]
Ok. I receive also Topic with the UID of the Module. This look exactly what i need, but the name of the value is confusing

my data from state/hostname/status when i change something
```
30.09.2023, 23:04:05 |          known topic | state/ac-dach/status with data {"protocol":"S21","online":true,"home":22.5,"heat":false,"fanrpm":0,"comp":0,"outside":13.0,"liquid":14.5,"power":false,"mode":"C","temp":21.0,"fan":"A","swingh":false,"swingv":false,"econo":false,"powerful":false,"comfort":false,"streamer":false,"sensor":false,"quiet":false,"autor":0.0,"autot":0.0,"auto0":"00:00","auto1":"00:00","autop":false}
```

with the data i received with the UID-Topic
```
30.09.2023, 23:04:05 |          known topic | 308AABBCCMYUID with data {"online":true,"target":21.00,"temp":22.50,"outside":13.00,"liquid":14.50,"comp":0,"fanrpm":0,"mode":"off","fan":"auto","swing":"off","preset":"home"}
```

| description | topic state/xyz/status | topic UID |
| ------------ | ------------------------ | ---- |
| temperature of the room | home | temp |
| set target temperature | temp | target |

is this intentional or a mistake?

**Reply by revk** [reply]
The first is the general status report. I think this one even has an array of [min,ave,max] for values covering the period when they are not fixed. This can be used with tools on linux to plot nice graphs. The manual should explain this.

The second is only sent when Home Assistant is enabled and is specifically values for HA as per its config. HA does not like being given arrays and just has the single snapshot values. This is also sent when changes happen not just on the minute.

And yes, the terminology was slightly different for HA.

**Comment by lorbetzki** [comment]
Ok, thank you for the explanation

**Comment by njh** [comment]
I asked a similar question in #74
I haven't had time to look at it recently, but when I last looked, I was unable to get reliable status updates over MQTT (but WebSockets periodic updates worked fine).

**Reply by revk** [reply]
There is a setting for live updates, should be in the manual.
